baremetal/metal3data_manager.go

```diff
@@ -479,9 +479,26 @@ func (m *DataManager) getAddressFromM3Pool(ctx context.Context, poolRef corev1.T
 	}
 
 	if !ipClaim.DeletionTimestamp.IsZero() {
-		// This Metal3IPClaim is about to be deleted so we cannot use it. Requeue.
-		m.Log.Info("Found IPClaim with deletion timestamp, requeuing.", "IPClaim", ipClaim)
-		return addressFromPool{}, true, nil
+		// Is it "our" ipClaim, or does it belong to an old and deleted Metal3Data with the same name?
+		our := false
```

Comment:
not a blocker but maybe we could rename it to something like: 
"owned" or "withOwnerRef"?

Reply:
yes this doesn't seem the right convention for the naming.

Reply:
Changed to `matchingOwnerRef`